src/server/graph-generation/interaction/index.js

```diff
@@ -56,20 +64,85 @@ function parse(data, queryIds){
     edges:[],
     nodes:[],
   };
-  let nodeMap=new Map(); //keeps track of nodes that have already been added
+  let nodeDegreeMap =new Map(); //keeps track of nodes that have already been added, also calculates node degree
   if(data){
     const dataSplit=data.split('\n\n');
     const nodeMetadata= new Map(dataSplit[1].split('\n').slice(1).map(line =>line.split('\t')).map(line => [line[0], line.slice(1) ]));
     dataSplit[0].split('\n').slice(1).forEach(line => {
       const splitLine=line.split('\t');
       const edgeMetadata = interactionMetadata(splitLine[6],splitLine[4]);
-      addInteraction([splitLine[0], splitLine[2]], splitLine[1], edgeMetadata, network, nodeMap, nodeMetadata, queryIds);
+      addInteraction([splitLine[0], splitLine[2]], splitLine[1], edgeMetadata, network, nodeDegreeMap, nodeMetadata, queryIds);
     });
-    return network;
+    return {network: network, nodeDegrees:nodeDegreeMap};
   }
   return {};
 }
 
+/**
+ * 
+ * @param {*} network JSON containing nodes and edges that represent a network
+ * @returns A network JSON with 50 nodes, sorted based on centrality & degree
+ * @description Each node in the network is assigned a `degree` metric, and the network is filtered down to the 100 nodes with largest degree.
+ * The remaining 100 nodes are assigned a `betweenness centrality (BC)` metric, and the network is filtered down to the 50 nodes with largest BC.
+ */
+function addMetricandFilter(network,nodeDegrees){
+  const nodes = network.nodes;
+  const edges = network.edges;
+
+  console.time("Sort/Filter Nodes");
+  //sort nodes by degree
+  let sortedNodes = nodes.sort( (a,b) => {
+    return nodeDegrees.get(b.data.id).length - nodeDegrees.get(a.data.id).length; 
+  });
+
+  //keep the 100 nodes with the largest degree
+  let filteredNodes = sortedNodes.slice(0,50);
+  filteredNodes.forEach( node => {
+    node.data['metric'] = nodeDegrees.get(node.data.id);
+  });
+  console.timeEnd("Sort/Filter Nodes");
+  
+  console.time('cy');
+  const cy = cytoscape( { headless:true, container:undefined, elements:{nodes:filteredNodes} } );
+  console.timeEnd('cy');
+
+  console.time('Filter Edges');
+  const filteredEdges = [];
+  edges.forEach( edge => {
+    const source = edge.data.source;
+    const target = edge.data.target;
+    let sourceFound = false;
+    let targetFound = false;
+    filteredNodes.forEach( node => {
```

Comment:
You could use a map rather than iterating over all nodes.  That's O(log n) rather than O(n).